File: platformplugin_win32.cpp
// Copyright (c) 2017-2025 The Khronos Group Inc.
//
// SPDX-License-Identifier: Apache-2.0

#include "pch.h"
#include "common.h"
#include "platformplugin.h"

#ifdef XR_USE_PLATFORM_WIN32
namespace {
struct Win32PlatformPlugin : public IPlatformPlugin {
    Win32PlatformPlugin(const Options*) { CHECK_HRCMD(CoInitializeEx(nullptr, COINIT_MULTITHREADED)); }

    virtual ~Win32PlatformPlugin() override { CoUninitialize(); }

    std::vector<std::string> GetInstanceExtensions() const override { return {}; }

    XrBaseInStructure* GetInstanceCreateExtension() const override { return nullptr; }

    void UpdateOptions(const Options* /*unused*/) override {}
};
}  // namespace

IPlatformPlugin* CreatePlatformPlugin_Win32(const Options* options) {
    return new Win32PlatformPlugin(options);
}
#endif

File: src/PlatformPluginWin32.zig
const Options = @import("Options.zig");
const xr = @import("openxr");

pub fn init(_: Options) @This() {
    return .{};
}

pub fn getInstanceExtensions(_: @This()) []const []const u8 {
    return &.{};
}

pub fn getInstanceCreateExtension(_: @This()) ?*xr.XrBaseInStructure {
    return null;
}

pub fn updateOptions(_: @This(), _: *Options) void {}
